codice/ddf
# DDF 2294  Refactor ValidationQueryFactory out of CachingFederationStrategy and into a new PreFederatedQueryPlugin

#### What does this PR do?
Refactor ValidationQueryFactory out of CachingFederationStrategy and into a new PreFederatedQueryPlugin as ValidationFilterPlugin
Also enforce ability to see error/warning record by user role

Currently,
1. The catalog framework can be configured to modify the query request at `ValidationQueryFactory` omit invalid data. 
2. Invalid metacard is also tagged post query by `MetacardValidityFilterPlugin` based on the user's role
3. FilterPlugin then remove the tagged metacard but the total count was not correctly updated

This ticket is to move step one logic into prefederatedQueryPlugin to modify the query request filter to allow or omit invalid data based on the the user's role. 
Downstream filters in step 2 and 3 remain unmodify

#### Who is reviewing it? 
@clockard 
@pklinef 
@millerw8 
@bdeining 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@clockard 

#### How should this be tested?
Use DDF image upload to upload an image twice
Reconfigure Duplication Validation configure to mark duplicate as error
Upload the same image again
Configure Validation Filter Plugin to ShowError and ShowWarning
Configure Metacard Validity to uncheck filterError and filterWarning
Use UI to search with * as admin user
Three records should be returned. One marked with warning and one with error
Logout and log back in as guest
UI search will only result in one record w/o warning or error
Logout and log back in as admin
Reconfigure Validation Filter Plugin to uncheck ShowWarning
UI Search only result in 2 records
Reconfigure Validation Filter plugin to uncheck ShowError
UI Search now only result in 1 record


#### Any background context you want to provide?
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2294
#### Screenshots

#### Checklist:
- [x ] Update / Add Unit Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
